shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala

```diff
@@ -132,27 +178,31 @@ class PickleReader {
       sym
     }
 
+    def readSymType(): Type =
+      try at(infoRef)(readType())
+      catch
+        case t: Throwable =>
+          throw new IllegalStateException(s"error while unpickling the type of $owner.$name", t)
+
     finishSym(tag match {
       case TYPEsym | ALIASsym =>
         var name1 = name.toTypeName
         // var flags1 = flags
         // if (flags.is(TypeParam)) flags1 |= owner.typeParamCreationFlags
         // newSymbol(owner, name1, flags1, localMemberUnpickler, privateWithin, coord = start)
-
-        // TODO: clsCtx at owner
-        clsCtx.createSymbol(name1, RegularSymbolFactory, addToDecls = false)
+        RegularSymbolFactory.createSymbol(name1, owner)
       case CLASSsym =>
         if isClassRoot then clsCtx.classRoot useWith { _.initialised = true }
         else if isModuleClassRoot then clsCtx.moduleClassRoot useWith { _.initialised = true }
-        else clsCtx.createSymbol(name.toTypeName, ClassSymbolFactory, addToDecls = false)
+        else ClassSymbolFactory.createSymbol(name.toTypeName, owner)
```

Comment:
I guess when it is not a top level class we should still add to decls of owner?

Reply:
Yes, indeed. Same as for Java classfiles inner classes. I added a comment.